### PR TITLE
[NodeTypeResolver] Handle hang on @template-extends in dependent Files

### PR DESCRIPTION
**Pull request by samsonasik**
Continue of https://github.com/rectorphp/rector-src/pull/1034 to avoid hang on `@template-extends` but on dependents file, see latest commit at https://github.com/boesing/rector-bug-template-extends/commit/9ebe4db5a1d7960c91a8f96e4b79757052ed83f7

on running : `vendor/bin/rector process src`

Closes https://github.com/rectorphp/rector/issues/6764

### CODE REVIEW

**Comment by samsonasik** [comment]
@boesing this should solve the hang on `@template-extends` in dependends files on https://github.com/boesing/rector-bug-template-extends/commit/9ebe4db5a1d7960c91a8f96e4b79757052ed83f7 .

I tried locally and it works 🎉 

Screen Shot 2021-10-22 at 22 04 59

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review. 

Same with https://github.com/rectorphp/rector-src/pull/1034 it only reproducible via repository.

To check this, it can be the following steps:

```bash
git clone https://github.com/boesing/rector-bug-template-extends
cd rector-bug-template-extends
composer install
```

Without this patch, running the following command will hang:

```bash
vendor/bin/rector process src
```

To test this patch, update composer.json as follow diff:

```diff
     },
     "require": {
-        "php": "~7.4.0",
+        "php": "^8.0",
         "myclabs/php-enum": "^1.8"
     },
     "require-dev": {
-        "rector/rector": "dev-main"
-    }
+        "rector/rector-src": "dev-handle-template-extends-in-dependends-file"
+    },
+    "minimum-stability": "dev",
+    "prefer-stable": true
```

then run composer update:

```bash
composer update
```

After that, run again `vendor/bin/rector process src --clear-cache`:

```bash
vendor/bin/rector process src --clear-cache
```

**Comment by samsonasik** [comment]
@TomasVotruba I added e2e workflow tests for it https://github.com/rectorphp/rector-src/pull/1044/commits/3aaf7e373a2540bf9d71681baea109fcd088a6f3 so we can ensure this patch is verified already in rector-src 🎉 🎉 🎉

File: e2e/enums/composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        "psr-4": {
+            "Whatever\\Foo\\Bar\\": "src/"
+        }
+    },
+    "require": {
+        "php": "^8.0",
+        "myclabs/php-enum": "^1.8"
+    },
+    "require-dev": {
+        "rector/rector-src": "dev-handle-template-extends-in-dependends-file"
```
Comment (samsonasik) [inline e2e/enums/composer.json:12]: note: this needs to be changed to `dev-main` once merged.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
This look very good :+1: we needed some kind of e2e tests for a while. Thank you @samsonasik :partying_face:

**Comment by boesing** [comment]
I would suggest renaming the e2e test to something related to `@template-extends` as you found out that it is not related to enums.

**Comment by samsonasik** [comment]
@boesing ok, renamed at https://github.com/rectorphp/rector-src/pull/1047

**Pull request merged**